http-client-core/src/main/java/io/micronaut/http/client/bind/binders/QueryValueClientArgumentRequestBinder.java

```diff
@@ -0,0 +1,400 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.client.bind.binders;
+
+import io.micronaut.core.annotation.AnnotationMetadata;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.beans.BeanProperty;
+import io.micronaut.core.beans.BeanWrapper;
+import io.micronaut.core.bind.annotation.Bindable;
+import io.micronaut.core.convert.ArgumentConversionContext;
+import io.micronaut.core.convert.ConversionContext;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.core.convert.format.Format;
+import io.micronaut.core.convert.format.FormattingTypeConverter;
+import io.micronaut.core.convert.value.ConvertibleMultiValues;
+import io.micronaut.core.convert.value.ConvertibleMultiValuesMap;
+import io.micronaut.core.convert.value.MutableConvertibleMultiValuesMap;
+import io.micronaut.core.type.Argument;
+import io.micronaut.http.MutableHttpRequest;
+import io.micronaut.http.annotation.QueryValue;
+import io.micronaut.http.client.bind.AnnotatedClientArgumentRequestBinder;
+import io.micronaut.http.client.bind.ClientRequestUriContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * Implementation of the Binder for {@link QueryValue}
+ * The details of implementation can be found in the
+ * {@link #bind(ArgumentConversionContext, ClientRequestUriContext, Object, MutableHttpRequest)} bind()} method javadoc.
+ *
+ * @author Andriy Dmytruk
+ * @since 3.0.0
+ */
+public class QueryValueClientArgumentRequestBinder implements AnnotatedClientArgumentRequestBinder<QueryValue> {
+    /**
+     * Values separated with commas ",". In case of iterables, the values are converted to {@link String} and joined
+     * with comma delimiter. In case of {@link Map} or a POJO {@link Object} the keys and values are alternating and all
+     * delimited with commas.
+     * <table borer="0">
+     *     <tr> <th><b> Type </b></th>      <th><b> Example value </b></th>                     <th><b> Example representation </b></th> </tr>
+     *     <tr> <td> Iterable &emsp </td>   <td> param=["Mike", "Adam", "Kate"] </td>           <td> "param=Mike,Adam,Kate" </td></tr>
+     *     <tr> <td> Map </td>              <td> param=["name": "Mike", "age": "30"] &emsp</td> <td> "param=name,Mike,age,30" </td> </tr>
+     *     <tr> <td> Object </td>           <td> param={name: "Mike", age: 30} </td>            <td> "param=name,Mike,age,30" </td> </tr>
+     * </table>
+     * Note that ambiguity may arise when the values contain commas themselves after being converted to String.
+     */
+    public static final String QUERY_VALUE_FORMAT_CSV = "CSV";
```

Comment:
I think these names should be lowercase, or at least match against what is put into the `@Format` annotation case insensitive